src/endpoint/endpoint.js

```diff
@@ -204,6 +208,16 @@ async function main(options = {}) {
     }
 }
 
+async function is_nsfs_http_denied(nsfs_config_root) {
+    if (!nsfs_config_root) {
```

Comment:
I think you can take the !nsfs_config_root condition outside instead of get into this flow.